lib/wanda_web/views/result_view.ex

```diff
@@ -15,7 +15,27 @@ defmodule WandaWeb.ResultView do
     render_one(result, ResultView, "result.json")
   end
 
-  def render("result.json", %{result: %ExecutionResult{payload: result, inserted_at: inserted_at}}) do
-    Map.put(result, "inserted_at", inserted_at)
+  def render("result.json", %{
+        result: %ExecutionResult{
+          execution_id: execution_id,
+          group_id: group_id,
+          status: status,
+          payload: result,
+          started_at: started_at
+        }
+      }) do
+    result
+    |> Map.put("started_at", started_at)
+    |> Map.put("execution_id", execution_id)
+    |> Map.put("group_id", group_id)
+    |> Map.put("result", extract_result(status, result))
+    |> Map.put("timeout", extract_timeout(status, result))
   end
+
+  # not really
+  defp extract_result(:running, _), do: :passing
```

Comment:
Doesn't this look strange? We are having a false passing here

Reply:
Yes it does :smile: 
And I am not really proud of the previous chain of `Map.put` either :sweat_smile: 

The fact is that we need to consider the running/completed variants of the executions also on the API layer.

I didn't go too far in that direction 'cause it might grow the PR, so I was thinking of just deferring to the next PRs to keep the scope narrow.

Honestly, I would completely remove the last commit https://github.com/trento-project/wanda/pull/60/commits/6656527aa23dbe55a5463071b0249aef04f44da4 for now

Reply:
At the end, as agreed, I simplified the factories here https://github.com/trento-project/wanda/pull/60/commits/7838b7c00059585a32112f02f5225e9a8d8aa4ad, yet this file will stay like this at the moment and be refactored in following PRs when we address the API layer